WalletWasabi/JsonConverters/SmartLabelJsonConverter.cs

```diff
@@ -3,24 +3,22 @@
 
 namespace WalletWasabi.JsonConverters;
 
-public class SmartLabelJsonConverter : JsonConverter
+public class SmartLabelJsonConverter : JsonConverter<SmartLabel>
 {
-	public override bool CanConvert(Type objectType)
-	{
-		return objectType == typeof(SmartLabel);
-	}
-
 	/// <inheritdoc />
-	public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+	public override SmartLabel? ReadJson(JsonReader reader, Type objectType, SmartLabel? existingValue, bool hasExistingValue, JsonSerializer serializer)
 	{
-		var s = reader.Value as string;
-		return new SmartLabel(s);
+		if (reader.Value is string serialized)
+		{
+			return new SmartLabel(serialized);
+		}
+		throw new ArgumentException($"No valid serialized {nameof(SmartLabel)} passed.");
```

Comment:
Generally I like better:

```suggestion

		throw new ArgumentException($"No valid serialized {nameof(SmartLabel)} passed.");
```

but I'm not sure what our guidelines say about this.

Reply:
Done.